circe/src/main/scala/org/http4s/circe/CirceInstances.scala

```diff
@@ -115,6 +115,12 @@ trait CirceInstances extends JawnInstances {
       Uri.fromString(str).leftMap(_ => "Uri")
     }
 
+  implicit class JsonDecoderSyntax[F[_]: JsonDecoder](private val req: Message[F])(
+      implicit F: JsonDecoder[F]) {
```

Comment:
Isn't it already specified as `F` context bound?
(also - don't you want to move this constraint to methods to make `implicit class` extends AnyVal to avoid allocation every time method is called?)

Reply:
Whoops. Yes. I switched how I was invoking it and left the old one. will fix these two.

Reply:
Can't extend AnyVal because of our position - 

`value class may not be a member of another class` - We could move these syntax's out but it would be a lot more involved move that I think I would rather do separately from this concept.